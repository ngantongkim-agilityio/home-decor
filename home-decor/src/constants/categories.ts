import {
  LivingRoomIcon,
  BedRoomIcon,
  DiningRoomIcon,
  KitchenIcon,
  OfficeIcon,
} from '@/components';

export const CATEGORIES_DATA = [
  {
    imageSrc: LivingRoomIcon,
    label: 'Living room',
    isActive: true,
  },
  {
    imageSrc: BedRoomIcon,
    label: 'Bed room',
    isActive: false,
  },
  {
    imageSrc: DiningRoomIcon,
    label: 'Dining room',
    isActive: false,
  },
  {
    imageSrc: KitchenIcon,
    label: 'Kitchen',
    isActive: false,
  },
  {
    imageSrc: OfficeIcon,
    label: 'Office',
    isActive: false,
  },
];
