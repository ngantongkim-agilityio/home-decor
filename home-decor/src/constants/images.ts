export const DEFAULT_IMAGE = require('@/assets/images/placeholder.png');

export const DEFAULT_AVATAR = require('@/assets/images/default-avatar.png');

export const BLUR_HASH =
  '|rF?hV%2WCj[ayj[a|j[az_NaeWBj@ayfRayfQfQM{M|azj[azf6fQfQfQIpWXofj[ayj[j[fQayWCoeoeaya}j[ayfQa{oLj?j[WVj[ayayj[fQoff7azayj[ayj[j[ayofayayayj[fQj[ayayj[ayfjj[j[ayjuayj[';
