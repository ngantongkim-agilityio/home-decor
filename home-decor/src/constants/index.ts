export * from './images';
export * from './slides';
export * from './api';
