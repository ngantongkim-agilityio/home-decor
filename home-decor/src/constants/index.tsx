export * from './images';
export * from './slides';
