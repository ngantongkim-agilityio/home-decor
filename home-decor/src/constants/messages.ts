export const ERROR_MESSAGES = {
  DEFAULT_ERROR: 'Something went wrong. Please try again later',
  INVALID_CREDENTIALS: 'Invalid credentials.',
  USER_EXISTS: 'User already exists with this email address',

  // Validate
  FIELD_REQUIRED: (fieldName: string) => `${fieldName} is required`,
  FIELD_INVALID: (fieldName: string) => `Invalid ${fieldName}`,
  INVALID_PASSWORD:
    'Password must be at least 8 characters and contain an uppercase letter, lowercase letter, and number',
  MISMATCH_PASSWORD: 'Passwords do not match',
};
