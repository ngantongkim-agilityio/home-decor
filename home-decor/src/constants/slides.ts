export const ONBOARDING_SLIDES = [
  {
    id: '1',
    title: 'Comfortable Space',
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
    image: require('@/assets/images/onboarding-step-one.png'),
  },
  {
    id: '2',
    title: 'Modern Design',
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
    image: require('@/assets/images/onboarding-step-two.png'),
  },
  {
    id: '3',
    title: 'Styled Living',
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
    image: require('@/assets/images/onboarding-step-three.png'),
  },
  {
    id: '4',
    title: 'Relaxing Furniture',
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
    image: require('@/assets/images/onboarding-step-four.png'),
  },
];
