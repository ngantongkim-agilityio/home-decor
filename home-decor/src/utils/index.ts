export * from './products';
export * from './notifications';
