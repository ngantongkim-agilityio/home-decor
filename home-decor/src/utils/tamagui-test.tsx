import { PropsWithChildren, ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { TamaguiProvider } from 'tamagui';
import { tamaguiConfig } from '@/themes/tamagui.config';

const Providers = ({ children }: PropsWithChildren) => {
  return <TamaguiProvider config={tamaguiConfig}>{children}</TamaguiProvider>;
};

const baseRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react-native';
export { baseRender as render };
