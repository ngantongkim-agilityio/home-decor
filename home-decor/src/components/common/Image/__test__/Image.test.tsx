import { render } from '@/utils/tamagui-test';
import { DEFAULT_IMAGE, BLUR_HASH } from '@/constants';
import { Image } from '..';

describe('Image Component', () => {
  const props = {
    testID: 'test-image',
  };

  it('Should render to match snapshot', () => {
    const component = render(<Image />);

    expect(component).toMatchSnapshot();
  });

  it('Should render with default props', () => {
    const { getByTestId } = render(<Image {...props} />);

    const image = getByTestId('test-image');

    expect(image.props.source).toEqual([DEFAULT_IMAGE]);
    expect(image.props.style).toEqual({
      width: 140,
      height: 160,
    });
    expect(image.props.placeholder).toEqual([{ BLUR_HASH }]);
  });
});
