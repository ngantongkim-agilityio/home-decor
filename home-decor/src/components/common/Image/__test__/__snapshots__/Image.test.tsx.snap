// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Component Should render to match snapshot 1`] = `
<View
  style={
    {
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      {
        "height": 160,
        "overflow": "hidden",
        "width": 140,
      }
    }
  >
    <ViewManagerAdapter_ExpoImage
      containerViewRef={"[React.ref]"}
      contentFit="cover"
      contentPosition={
        {
          "left": "50%",
          "top": "50%",
        }
      }
      height={160}
      nativeViewRef={"[React.ref]"}
      onError={[Function]}
      onLoad={[Function]}
      onLoadStart={[Function]}
      onProgress={[Function]}
      placeholder={
        [
          {
            "BLUR_HASH": "|rF?hV%2WCj[ayj[a|j[az_NaeWBj@ayfRayfQfQM{M|azj[azf6fQfQfQIpWXofj[ayj[j[fQayWCoeoeaya}j[ayfQa{oLj?j[WVj[ayayj[fQoff7azayj[ayj[j[ayofayayayj[fQj[ayayj[ayfjj[j[ayjuayj[",
          },
        ]
      }
      source={
        [
          {
            "testUri": "../../../assets/images/placeholder.png",
          },
        ]
      }
      style={
        {
          "height": 160,
          "width": 140,
        }
      }
      transition={null}
      width={140}
    />
  </View>
</View>
`;
